lms/djangoapps/email_marketing/signals.py

```diff
@@ -199,6 +210,35 @@ def email_marketing_user_field_changed(sender, user=None, table=None, setting=No
             return
         update_user_email.delay(user.email, old_value)
 
+@receiver(post_save, sender=Site)
+def email_marketing_new_site_added(sender, **kwargs):
+    """
+    update email marketing config on the addition of new site.
+    """
+    instance = kwargs['instance']
+
+    email_config = EmailMarketingConfiguration.current()
+    if not email_config.enabled:
+        return
+
+    user_lists = json.loads(email_config.sailthru_user_list)
+    if not user_lists.get(instance.domain):
+        update_email_marketing_config_sailthru_user_list.delay(instance.domain)
+
+
+@receiver(post_save, sender=EmailMarketingConfiguration)
+def email_marketing_configuration_list(sender, **kwargs):
+    """
+    On creation of new email marketing configuration check if new sailthru
+    list is added and create corresponding list in sailthru.
+    """
+    email_config = kwargs['instance']
+    if not email_config.enabled:
+        return
+
+    user_lists = json.loads(email_config.sailthru_user_list)
+    create_sailthru_user_list.delay(user_lists)
```

Comment:
Now I see, you are not deleting the existing lists. If there is a name change on the site, you will create a new list. That's good.